test/upgrade/autoscaler.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"time"
+
+	"go.uber.org/zap"
+
+	"golang.org/x/sync/errgroup"
+	"knative.dev/serving/test"
+
+	pkgupgrade "knative.dev/pkg/test/upgrade"
+
+	"knative.dev/serving/pkg/apis/autoscaling"
+	rtesting "knative.dev/serving/pkg/testing/v1"
+	"knative.dev/serving/test/e2e"
+)
+
+const (
+	containerConcurrency = 6
+	targetUtilization    = 0.7
+)
+
+// AutoscaleSustainingTest checks that when traffic increases a knative app
+// scales up and sustains the scale as long as the traffic sustains, despite whether
+// it is switching modes between normal and panic.
+func AutoscaleSustainingTest(logger *zap.Logger) pkgupgrade.BackgroundOperation {
+	var ctx *e2e.TestContext
+	var grp errgroup.Group
+	stopCh := make(chan time.Time)
+	return pkgupgrade.NewBackgroundVerification("AutoscaleSustainingTest",
+		func(c pkgupgrade.Context) {
+			// Setup
+			ctx = e2e.SetupSvc(c.T, autoscaling.KPA, autoscaling.Concurrency, containerConcurrency, targetUtilization,
+				rtesting.WithConfigAnnotations(map[string]string{
+					autoscaling.TargetBurstCapacityKey: "0", // Not let Activator in the path.
+				}))
+			grp.Go(func() error {
+				return e2e.AssertAutoscaleUpToNumPods(ctx, logger.Sugar().Infof, 1, 10, stopCh, false /* quick */)
```

Comment:
Aren't we like to control those parameters: `1, 10, false`?

Reply:
They're constants. I'll just move them up to the `const` declaration.